cmd/eval-dev-quality/cmd/evaluate.go

```diff
@@ -414,3 +420,25 @@ func writeCSVs(resultPath string, assessments report.AssessmentPerModelPerLangua
 
 	return nil
 }
+
+// withLoadedModel loads the model for the duration of the given function if supported by the model's provider.
+func withLoadedModel(log *log.Logger, model model.Model, providerForModel map[model.Model]provider.Provider, function func()) {
```

Comment:
We do have the `model` already so no need to pass the whole map, would only pass the single Provider.

Reply:
If you are scared about having no entry on that key, it is impossible as the list of models is also built when the list of providers is built. 

Should we encounter an no key error than we are manipulating the list of models and having one without a provider is in itself highly illegal.

Reply:
Yeeeahhhh.... I was thinking about that but... I guess we can risk it?